src/main.rs

```diff
@@ -221,12 +295,53 @@ where
     writeln!(file, "source=(\"{}\")", source)?;
     writeln!(file, "sha256sums=(\"{}\")", sha256)?;
     writeln!(file)?;
+    // Include the prepare, build and check steps for non-binary package.
+    if no_bin {
+        writeln!(file, "prepare() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(
+            file,
+            "    cargo fetch --locked --target \"$(rustc -vV | sed -n 's/host: //p')\""
+        )?;
+        writeln!(file, "}}")?;
+        writeln!(file)?;
+        writeln!(file, "build() {{")?;
+        writeln!(file, "    cd $pkgname-$pkgver")?;
+        writeln!(file, "    export RUSTUP_TOOLCHAIN=stable")?;
+        writeln!(file, "    export CARGO_TARGET_DIR=target")?;
+        writeln!(file, "    cargo build --frozen --release --all-features")?;
```

Comment:
Should we be assuming that they want `--all-features` by default for the release build?

Reply:
I got this from the Arch wiki, so I was assuming this would be a sensible default. Future enhancement could be to further specialise? https://wiki.archlinux.org/title/Rust_package_guidelines#Build

Reply:
I'd say let's leave it out for now and consider it for future enhancement. In such a case I think it's reasonable to wait until someone asks for it - otherwise adding it where it _isn't_ wanted could cause trouble, as in [the case of Aura](https://github.com/fosskers/aura/blob/master/rust/aura-pm/Cargo.toml).